bokeh/util/token.py

```diff
@@ -232,7 +227,22 @@ def check_session_id_signature(session_id: str,
 # Private API
 #-----------------------------------------------------------------------------
 
-def _get_sysrandom() -> Tuple[Any, bool]:
+class _BytesEncoder(json.JSONEncoder):
+    def default(self, x):
+        if isinstance(x, bytes):
+            return dict(bytes=_base64_encode(x))
+        return super().default(x)
+
+class _BytesDecoder(json.JSONDecoder):
+    def __init__(self, *args, **kwargs):
+        super().__init__(object_hook=self.bytes_object_hook, *args, **kwargs)
+
+    def bytes_object_hook(self, obj):
+        if set(obj.keys()) == {"bytes"}:
+            return _base64_decode(obj["bytes"])
+        return obj
```

Comment:
Perhaps using bokeh's serializer would be a better choice than implementing an ad-hoc one. bokeh's serializer can be used outside the context of model serialization and supports `bytes` and pretty much everything else out of the box.

Reply:
@mattpap I don't actually see how to apply these, and there are no docs
```
In [10]: data
Out[10]: b'foo'

In [11]: s.serialize(data)
Out[11]: Serialized(content={'type': 'bytes', 'data': Buffer(id='1002', data=b'foo')}, buffers=[Buffer(id='1000', data=b'foo'), Buffer(id='1001', data=b'foo'), Buffer(id='1002', data=b'foo')])

In [12]: s.encode(data)
Out[12]: {'type': 'bytes', 'data': Buffer(id='1003', data=b'foo')}
```
neither of those outputs can be returned as-is from a JSON encoder/decoder. If you want to push commits to this PR to update 👍 otherwise absent any statements, I will plan to merge as-is tomorrow since all tests are passing.

Reply:
> (...) with four copies of the data, actually what is intended?

It depends what you did previously. The snippet is incomplete, but I will assume that multiple calls to `serialize()` and/or `encode()` were executed. Then it's correct for an instance of `Serializer` to collect all objects from all invocations. This means you must create new instance whenever serializing unrelated data. Multiple calls to serialization methods are allowed, but they work, by design, with the existing state of the serializer (or deserializer for that matter). This way you can serialize multiple related data streams, without having to maintain state separately, as we used to do.

Reply:
>  (...) nor do I see what would just turn them to/from strs without using json module

Correct, this is serialization and not encoding. You need an additional step to turn the intermediate representation generated by `Serializer` to something that you can actually send over the wire. In bokeh's case this can be pure JSON, but also multi-part JSON and binary (that we use with the server). Though it should be possible to use YAML, etc. as well.

Reply:
Documentation would be nice to have.

Reply:
@mattpap I've never encountered this usage of terminology before, in ~25 year of development. Serialization has always and only ever implied serialization *to bytes* (or *to string*) in every context I have seen it. I've never once seen "serialization" used to mean "conversion to some intermediate data structure. FWIW references seem to agree with this [[1]](https://en.wikipedia.org/wiki/Serialization) [[2]](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/serialization/) and I think #12232 is further evidence that this behaviour is not the expected behaviour of something that "serializes". I would actually suggest that we think about aligning the terminology with common usage and find some other name for this intermediate step, and reserve "serialization" only for actions that return bytes and/or strings.

Reply:
In any case if you can point at how to actually get a string back, I'll happily just remove the ad-hoc bytes encoder.